tests/e2e/targetallocator-prometheuscr/00-install.yaml

```diff
@@ -107,6 +107,9 @@ spec:
     image: "local/opentelemetry-operator-targetallocator:e2e"
     prometheusCR:
       enabled: true
+    observability:
+      metrics:
+        enableMetrics: true
```

Comment:
shall we assert that the service monitor is created?

Reply:
@changexd since we have e2e test it would be good to add assertion on the created service monitor.